NebulaAPI/Extensions/PlayerCollectionExtensions.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NebulaAPI.GameState;
+using NebulaAPI.Networking;
+
+namespace NebulaAPI.Extensions;
+
+public static class PlayerCollectionExtensions
+{
+    // public static IReadOnlyCollection<IPlayerData> GetAllPlayerDataIncludingHost(
+    //     this IReadOnlyCollection<INebulaPlayer> players, ILocalPlayer host)
+    // {
+    //     var all = players
+    //         .Select(p => p.Data)
+    //         .ToList();
+    //     all.Add(host.Data);
+    //
+    //     return all;
+    // }
+
+    public static IReadOnlyDictionary<INebulaConnection, INebulaPlayer> GetByConnectionStatus(
+        this IReadOnlyDictionary<INebulaConnection, INebulaPlayer> playerConnections, EConnectionStatus status)
+    {
+        return playerConnections
+            .Where(kvp => kvp.Key.ConnectionStatus == status)
+            .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+    }
+
+    public static IReadOnlyCollection<INebulaPlayer> GetByConnectionStatus(
+        this IReadOnlyCollection<INebulaPlayer> players, EConnectionStatus status)
+    {
+        return players
+            .Where(value => value.Connection.ConnectionStatus == status)
+            .ToList();
+    }
+
+    public static INebulaPlayer GetByPlayerId(
+        this IReadOnlyCollection<INebulaPlayer> players, ushort id)
+    {
+        return players.First(p => p.Id == id);
+    }
```

Comment:
This will throw an exception if there is no player found with the given ID. Should we do this or do FirstOrDefault and just return null?